### DISCUSSION TITLE
Responsive plotting

### DISCUSSION THREAD

**Discussion by MattOslin** [foxglove/community#89] [Q&A]
Hello, I would like to build a plot similar to ones I've made in the past where dragging a slider updates a plot live. My two use cases for this are for after sending a single message, or while in an opened file and paused. I can sort of get the behavior I'd like today by reader the slider in a user script, but the user script is only run when a new message comes in, not when the slider value is updated.

**Comment by defunctzombie** [comment]
@jtbandes I started writing a response to this around how this is expected behavior and a limitation around how we process messages and time but it occurs to me that it's not that different to a user making changes to a user script while paused. I think we do run a script in that case? So maybe we could also run on global variable updates?

**Reply by amacneil** [reply]
I'm pretty sure we used to rerun on global variable updates - that was the whole point of the Variable Slider panel

**Reply by defunctzombie** [reply]
https://github.com/foxglove/studio/issues/4989

**Reply by jtbandes** [reply]
🤷 the variable slider updates the variable. Panels see the new variable. Whether that makes user scripts run or not, I can’t recall. When I glanced at the code I did not see any attempt to rerun scripts when variables change. So doubt this is a regression, but a good ask. However, do we also rerun the script on all preloaded blocks?

**Reply by jtbandes** [reply]
I was wrong, it is a regression 😕 https://github.com/foxglove/studio/issues/4989#issuecomment-1355755318
